src/account/UpgradeableModularAccount.sol

```diff
@@ -251,11 +253,13 @@ contract UpgradeableModularAccount is
 
         // Run any pre plugin exec specific to this caller and the `executeFromPluginExternal` selector
 
-        PostExecToRun[] memory postPermittedCallHooks =
-            _doPrePermittedCallHooks(IPluginExecutor.executeFromPluginExternal.selector, msg.sender);
+        PostExecToRun[] memory postPermittedCallHooks = _doPrePermittedCallHooks(
+            getPermittedCallKey(msg.sender, IPluginExecutor.executeFromPluginExternal.selector), msg.data
+        );
 
         // Run any pre exec hooks for this selector
-        PostExecToRun[] memory postExecHooks = _doPreExecHooks(IPluginExecutor.executeFromPluginExternal.selector);
+        PostExecToRun[] memory postExecHooks =
+            _doPreExecHooks(IPluginExecutor.executeFromPluginExternal.selector, msg.data);
```

Comment:
ditto on the usage of `msg.data`

Reply:
Same reasoning as above for permitted call hooks - the hook function benefits from the standardized format, and gets access to the target address and the value, which are otherwise not visible if we just send `data`.